src/particles.zig

```diff
@@ -365,91 +379,197 @@ pub const DirectionMode = union(enum(u8)) {
 	// The particle goes in a random direction
 	scatter: void,
 	// The particle goes in the specified direction
-	direction: Vec3f,
+	direction: DirectionData,
+
+	pub const DirectionData = struct {
```

Comment:
Please move this cone-based direction mode into a separate PR, I'd rather not review it right now, there are too many changes in this one file already.